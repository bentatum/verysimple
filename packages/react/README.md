# @verysimple/react

This is the `@verysimple/react` package