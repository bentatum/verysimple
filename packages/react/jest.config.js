/** @type {import('jest').Config} */
module.exports = {
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect"],
  collectCoverage: true,
  coverageThreshold: {
    global: {
      statements: 75,
      branches: 72,
      functions: 75,
      lines: 75,
    },
  },
  coverageReporters: ["text-summary", "json-summary"],
  collectCoverageFrom: ["src/**/*.{ts,tsx}", "!**/*.d.ts"],
  coverageDirectory: "coverage",
  moduleNameMapper: {
    "@/test/(.*)": "<rootDir>/test/$1",
    "@/(.*)$": "<rootDir>/src/$1",
  },
  modulePathIgnorePatterns: ["<rootDir>/dist/"],
  prettierPath: null,
};
