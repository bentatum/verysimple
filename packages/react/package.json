{
  "name": "@verysimple/react",
  "version": "8.1.1",
  "description": "A very simple component library",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "author": "Ben Tatum <mail@benjtatum.com>",
  "license": "MIT",
  "repository": "https://github.com/bentatum/verysimple",
  "scripts": {
    "test": "jest",
    "prebuild": "rm -rf dist",
    "build": "tsc --project tsconfig.build.json && tsc-alias"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.20.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/inquirer": "latest",
    "@types/jest": "^29.2.3",
    "@types/react": "^18.0.21",
    "@types/testing-library__jest-dom": "^5.14.5",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "classnames": "^2.3.2",
    "framer-motion": "^10.15.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.1.8",
    "tsc-alias": "^1.8.2",
    "typescript": "^4.9.3"
  },
  "volta": {
    "node": "20.10.0",
    "yarn": "1.22.21"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.9",
    "@heroicons/react": "^2.0.13",
    "clsx": "^2.1.0",
    "framer-motion": "^10.15.0"
  }
}
