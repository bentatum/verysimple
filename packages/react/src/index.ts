export * from "./ButtonBase";
export * from "./Card";
export * from "./Dialog";
export * from "./FilledButtonBase";
export * from "./FilledIconButton";
export * from "./GradientBorder";
export * from "./helpers";
export * from "./IconButton";
export * from "./NormalIconButton";
export * from "./OutlinedButtonBase";
export * from "./OutlinedIconButton";
export * from "./Spinner";
export * from "./types";
