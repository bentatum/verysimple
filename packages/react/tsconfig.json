{
  "extends": "@verysimple/tsconfig/react-library.json",
  "compilerOptions": {
    "outDir": "./dist",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules"]
}
