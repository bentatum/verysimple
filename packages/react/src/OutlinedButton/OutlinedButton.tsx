import clsx from "clsx";
import { forwardRef } from "react";
import type { ButtonColor } from "@/types";
import type { ButtonBaseProps } from "@/ButtonBase";
import { fieldPaddingClassNames } from "@/helpers/fieldPaddingClassNames";
import OutlinedButtonBase from "@/OutlinedButtonBase";

export interface OutlinedButtonProps extends ButtonBaseProps {
  color?: ButtonColor;
}

export const OutlinedButton = forwardRef<
  HTMLButtonElement,
  OutlinedButtonProps
>(
  (
    {
      color = "neutral",
      disabled,
      size = "md",
      children,
      as = "button",
      ...props
    },
    ref
  ) => {
    return (
      <OutlinedButtonBase
        as={as}
        ref={ref}
        size={size}
        color={color}
        disabled={disabled}
        {...props}
      >
        {children}
      </OutlinedButtonBase>
    );
  }
);

OutlinedButton.displayName = "OutlinedButton";

export default OutlinedButton;
