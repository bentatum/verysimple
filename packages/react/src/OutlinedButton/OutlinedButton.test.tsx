import { render, screen } from "@testing-library/react";
import { createRef } from "react";
import OutlinedButton from "./OutlinedButton";

describe("OutlinedButton", () => {
  it("should have an outline", () => {
    const { container } = render(<OutlinedButton>Test</OutlinedButton>);
    expect(screen.getByRole("button")).toHaveClass("border", "my-border")
    expect(container).toMatchSnapshot();
  });

  it("can render as a custom component", () => {
    const CustomComponent = () => <div data-testid="custom-component" />;
    const { container } = render(
      <OutlinedButton as={CustomComponent}>Test</OutlinedButton>
    );
    expect(screen.getByTestId("custom-component")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("can render as a custom component when primary", () => {
    const CustomComponent = () => <div data-testid="custom-component" />;
    const { container } = render(
      <OutlinedButton as={CustomComponent} color="primary">
        Test
      </OutlinedButton>
    );
    expect(screen.getByTestId("custom-component")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("can render as a custom component when primary and disabled", () => {
    const CustomComponent = () => <div data-testid="custom-component" />;
    const { container } = render(
      <OutlinedButton as={CustomComponent} color="primary" disabled>
        Test
      </OutlinedButton>
    );
    expect(screen.getByTestId("custom-component")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("forwards the ref", () => {
    const ref = createRef<HTMLButtonElement>();
    render(<OutlinedButton ref={ref}>test</OutlinedButton>);
    expect(ref.current).toBeInTheDocument();
  });
  it("has a display name", () => {
    expect(OutlinedButton.displayName).toBe("OutlinedButton");
  });
});
