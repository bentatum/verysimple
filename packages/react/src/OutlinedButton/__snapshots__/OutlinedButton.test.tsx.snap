// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlinedButton can render as a custom component 1`] = `
<div>
  <div
    data-testid="custom-component"
  />
</div>
`;

exports[`OutlinedButton can render as a custom component when primary 1`] = `
<div>
  <div
    data-testid="custom-component"
  />
</div>
`;

exports[`OutlinedButton can render as a custom component when primary and disabled 1`] = `
<div>
  <div
    data-testid="custom-component"
  />
</div>
`;

exports[`OutlinedButton should have an outline 1`] = `
<div>
  <button
    class="focus:outline-none cursor-pointer disabled:cursor-not-allowed inline-flex items-center justify-center box-content shadow-sm focus:shadow-md hover:shadow-md disabled:shadow-none transition-shadow ease-in duration-100 disabled:text-zinc-400 dark:disabled:text-zinc-500 whitespace-nowrap font-bold rounded-full text-base min-h-[48px] border my-border-primary hover:my-bg-gradient-to-r hover:border-none hover:p-[1px] hover:text-white focus:my-bg-gradient-to-r focus:border-none focus:p-[1px] focus:text-white"
    tabindex="0"
  >
    <span
      class="px-5 w-full flex items-center justify-center"
    >
      Test
    </span>
  </button>
</div>
`;
